packages/python/plotly/plotly/tests/test_core/test_graph_objs/test_layout_subplots.py

```diff
@@ -39,7 +39,7 @@ def test_initial_access_subplot2(self):
             self.layout.xaxis2
 
     def test_initial_access_subplot2(self):
-        with pytest.raises(KeyError):
```

Comment:
why this change? I think `KeyError` is the appropriate error here, no?

Reply:
The way the better error messages work for subclasses of `BasePlotlyType` is they recursively call `BasePlotlyType.__getitem__(last_prop,next_prop_name)` where `last_prop` is the object resulting from the prop just looked up in the list of properties (the split magic underscores list) and `next_prop_name` is the next property in the list to look up. If we raise KeyError at the end of this list, we don't get the nice print out of all the possible properties like you do with `BasePlotlyType._raise_on_invalid_property_error`, which raises `ValueError`. Now we could change that to `KeyError` but I wouldn't recommend it because: a) we will break a lot more tests (my change to make it throw `ValueError` resulted in only this test breaking :+1:) and b) we won't get a nice print out of the error message, because the Python interpreter calls `repr` on the messages of `KeyError` so that `{}['']` prints `'KeyError: ""` and not `KeyError: ` (see e.g., https://stackoverflow.com/questions/34051333/strange-error-message-printed-out-for-keyerror). That's probably why `BasePlotlyType._raise_on_invalid_property_error` raises `ValueError` in the first place :-)

Reply:
This would be a breaking change. As I mentioned above, we should be able to subclass `KeyError` to customize the displayed string and this would not be breaking.

Reply:
Ok that works :+1:. Does this work for the subclass:
```python
class PlotlyKeyError(KeyError):
    def __str__(self):
        return Exception.__str__(self)
```